### ISSUE TITLE
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\ensemble\weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.   from numpy.core.umath_tests import inner1d

### ISSUE THREAD

**Issue by mounaBoo** [numpy/numpy#13568]
C:\ProgramData\Anaconda3\lib\site-packages\sklearn\ensemble\weight_boosting.py:29: DeprecationWarning: numpy.core.umath_tests is an internal NumPy module and should not be imported. It will be removed in a future NumPy release.
  from numpy.core.umath_tests import inner1d

`import numpy as np
import pandas as pd
from sklearn.kernel_approximation import RBFSampler
from sklearn.linear_model import SGDClassifier
from sklearn.model_selection import train_test_split
from sklearn import svm
from sklearn.metrics import classification_report
from sklearn import metrics
from sklearn.linear_model import LogisticRegression
from sklearn.naive_bayes import GaussianNB
from sklearn.neighbors import KNeighborsClassifier
from sklearn.tree import DecisionTreeClassifier
from sklearn.metrics import (precision_score, recall_score,f1_score, accuracy_score,mean_squared_error,mean_absolute_error)
from sklearn.ensemble import AdaBoostClassifier
from sklearn.ensemble import RandomForestClassifier
from sklearn.preprocessing import Normalizer
from sklearn.model_selection import GridSearchCV
from sklearn.svm import SVC
from sklearn.metrics import confusion_matrix
from sklearn.metrics import (precision_score, recall_score,f1_score, accuracy_score,mean_squared_error,mean_absolute_error, roc_curve, classification_report,auc)

from sklearn.preprocessing import OneHotEncoder, LabelEncoder`

**Comment by bashtage** [comment]
This is a scikit-learn issue since they are using a module in a way that they should not.

**Comment by mhvk** [comment]
@mounaBoo - does `skikit-learn` use `inner1d`? If so, how? It was meant only as an example custom ufunc, tested to ensure that we do not accidentally break ufunc creation. But if it does something beyond the regular `np.inner`, perhaps we should expose it...

**Comment by charris** [comment]
This issue has come up before, don't recall exactly when.

**Comment by ksunden** [comment]
This has been fixed in scikit-learn https://github.com/scikit-learn/scikit-learn/pull/10880

See also #10815

**Comment by seberg** [comment]
Sounds like scikit-learn has fixed the issue on their end, and that is fine for the moment. If someone wants `inner1d` exposed, please open a new issue for that. I will close this one. Thanks all.
